libs/s25main/ai/aijh/AIPlayerJH.cpp

```diff
@@ -1369,9 +1369,17 @@ void AIPlayerJH::HandleBorderChanged(const MapPoint pt)
     const auto* mil = gwb.GetSpecObj<nobMilitary>(pt);
     if(mil)
     {
-        if(mil->GetFrontierDistance() != FrontierDistance::Far && mil->IsGoldDisabled())
+        if(mil->GetFrontierDistance() != FrontierDistance::Far)
         {
-            aii.SetCoinsAllowed(pt, true);
+            if(mil->IsGoldDisabled())
+                aii.SetCoinsAllowed(pt, true);
+
+            // Fill up with soldiers
+            for(unsigned rank = 0; rank < NUM_SOLDIER_RANKS; ++rank)
+            {
+                if(mil->GetDesiredTroops(rank) != mil->GetMaxTroopsCt())
+                    aii.SetDesiredTroops(mil->GetPos(), rank, mil->GetMaxTroopsCt());
```

Comment:
What is this supposed to do? Doesn't this set the desired troops such that each rank should be there fully filling the building up? I.e. wouldn't this request for a building with capacity for 2 soldiers, 2 soldiers of each rank, i.e. 2*NUM_SOLDIER_RANKS soldiers overfilling the building?

Or is "desired_troops" rather a limit on how many soldiers of that rank are desired? Maybe a "max" in the name would help then.